### PR TITLE
ES|QL: Fix stats by constant expression

### LINKED ISSUE #114714: ESQL: STATS BY expression not working with expression from EVAL
`FROM employees | eval x = [1,2,3], y = 5 + 6 | stats by y + 1` or `FROM employees | eval x = [1,2,3], y = 5 + 6 | stats by yy = y + 1` currently return 0 rows, even though using expressions in the `by` clause works fine if they use only fields that are present in the index. Using `by` clauses composed from both index fields and eval fields seems to work, too, like `FROM test | eval x = [1,2,3], y = 5 + 6 | stats by yy = y + a`.

### PR DESCRIPTION
**Pull request by luigidellaquila**
Fixes: https://github.com/elastic/elasticsearch/issues/114714

Do not consider implicit aliases in `Aggregate.computeReferences()`. 
These aliases could be misleading for the field_caps to compute involved indices.

### CODE REVIEW

**Comment by luigidellaquila** [comment]
@elasticmachine update branch

File: x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec [hunk 1]
```diff
@@ -2473,3 +2473,13 @@ c2:l |c2_f:l |m2:i |m2_f:i |c:l
 1    |0      |2    |2      |19
 1    |1      |5    |5      |21
 ;
+
+
+statsByConstantExpression
+required_capability: fix_stats_by_foldable_expression
+FROM employees | eval x = [1,2,3], y = 5 + 6 | stats by y+1
```
Comment (alex-spies) [inline x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec:2480]: nit: we could also add the example with the explicit alias in the group `STATS BY yy = y+1`. And test cases with a non-empty list of aggs.

**Review by alex-spies — APPROVED** [review]
LGTM, thanks Luigi!

**Comment by luigidellaquila** [comment]
@elasticmachine update branch

**Pull request merged**